src/plugins/dist_matrixops/dist_constant.cpp

```diff
@@ -45,8 +44,8 @@ namespace phylanx { namespace dist_matrixops { namespace primitives
                 constant_d(
                     _1_value,
                     _2_shape,
-                    __arg(_3_tile_index, nil),
-                    __arg(_4_numtiles, nil),
+                    __arg(_3_tile_index, find_here()),
+                    __arg(_4_numtiles, num_localities()),
```

Comment:
Wow, I would never have thought of using a primitive invocation as the default value for an argument - nice!